File: sqlbuilder/service/AutoIncService.java
package com.clickability.cms.dataaccess.service;

/**
 * Basic CRUD operations for
 * the most common case: an integer primary key
 */
public interface AutoIncService<T> extends DbService<T> {
	
	T findById(int id) throws Exception;

	void delete(int id) throws Exception;
	
}

File: queryBuilder/src/main/kotlin/queryBuilder/com/inunison/lib/queryBuilder/service/AutoIncService.kt
package com.clickability.cms.dataaccess.service

/**
 * Basic CRUD operations for
 * the most common case: an integer primary key
 */
interface AutoIncService<T> : DbService<T> {

    @Throws(Exception::class)
    fun findById(id: Int): T

    @Throws(Exception::class)
    fun delete(id: Int)

}
